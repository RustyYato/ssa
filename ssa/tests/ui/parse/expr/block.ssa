let with_expr = {
    let y = x;
    x
};
let no_expr = {
    let y = x;
};
