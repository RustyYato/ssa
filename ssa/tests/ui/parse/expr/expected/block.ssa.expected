[
  Item {
    id: ItemId(1)
    kind: ItemKind::Let(Let {
        binding: Ident {
          id: IdentId(1)
          name: "with_expr"
        }
        ty: 
        value: Option::Some(Expr {
            id: ExprId(3)
            kind: ExprKind::Block(Block {
                stmts: [
                  Stmt {
                    id: StmtId(1)
                    kind: StmtKind::Let(Let {
                        binding: Ident {
                          id: IdentId(2)
                          name: "y"
                        }
                        ty: 
                        value: Option::Some(Expr {
                            id: ExprId(1)
                            kind: ExprKind::Ident(Ident {
                                id: IdentId(3)
                                name: "x"
                              }
                            )
                          }
                        )
                      }
                    )
                  }
                ]
                expr: Option::Some(Expr {
                    id: ExprId(2)
                    kind: ExprKind::Ident(Ident {
                        id: IdentId(4)
                        name: "x"
                      }
                    )
                  }
                )
              }
            )
          }
        )
      }
    )
  }
  Item {
    id: ItemId(2)
    kind: ItemKind::Let(Let {
        binding: Ident {
          id: IdentId(5)
          name: "no_expr"
        }
        ty: 
        value: Option::Some(Expr {
            id: ExprId(5)
            kind: ExprKind::Block(Block {
                stmts: [
                  Stmt {
                    id: StmtId(3)
                    kind: StmtKind::Let(Let {
                        binding: Ident {
                          id: IdentId(6)
                          name: "y"
                        }
                        ty: 
                        value: Option::Some(Expr {
                            id: ExprId(4)
                            kind: ExprKind::Ident(Ident {
                                id: IdentId(7)
                                name: "x"
                              }
                            )
                          }
                        )
                      }
                    )
                  }
                ]
                expr: 
              }
            )
          }
        )
      }
    )
  }
]
